test.c

```diff
@@ -1839,6 +1866,32 @@ response_status_cb (http_parser *p, const char *buf, size_t len)
   return 0;
 }
 
+int
+chunk_header_cb (http_parser *p)
+{
+  assert(p == parser);
+  if (p->content_length == 0) {
+    // Terminating chunk.  Don't increment num_chunks in this case.
```

Comment:
True, I didn't understand this part fully either. In the tests I see stuff like
```
,.num_chunks= 0
,.num_chunks_complete= 1
```
but it looks to me like num_chunks_complete is always one larger than num_chunks, so whats the point of having both? Let me see if I can collapse these to one, checking with the original proxygen changeset's author if necessary...